tensorboard/plugins/debugger_v2/debug_data_multiplexer.py

```diff
@@ -229,15 +234,29 @@ def Alerts(self, run, begin, end):
             else:
                 alert_type = "__MiscellaneousAlert__"
             alerts_breakdown[alert_type] = len(monitor_alerts)
-        end = self._checkBeginEndIndices(begin, end, len(alerts))
-        # TODO(cais): Add support for filtering by alert type.
+            alerts_by_type[alert_type] = monitor_alerts
+
+        if alert_type_filter is not None:
+            if alert_type_filter not in alerts_breakdown:
+                raise errors.InvalidArgumentError(
+                    "Filtering of alerts failed: alert type %s does not exist"
+                    % alert_type_filter
+                )
+            end = self._checkBeginEndIndices(
+                begin, end, alerts_breakdown[alert_type_filter]
+            )
```

Comment:
Can you deduplicate this by checking against `alert_list` after the
`if`-statement?

Reply:
Done.